src/main/java/org/jfrog/hudson/pipeline/common/types/buildInfo/BuildInfo.java

```diff
@@ -294,6 +301,51 @@ private void addModule(Module other) {
             currentModule.append(other);
         }
     }
+    public void appendDependencies(List<Dependency> dependencies, String moduleId) {
+        Module defaultModule = new ModuleBuilder()
+                .id(moduleId)
+                .dependencies(dependencies)
+                .build();
+        Module currentModule = this.getModules().stream()
+                // Check if the default module already exists.
+                .filter(module -> StringUtils.equals(module.getId(), moduleId))
+                .findAny()
+                .orElse(null);
+        if (currentModule != null) {
+            currentModule.append(defaultModule);
+        } else {
+            this.getModules().add(defaultModule);
+        }
+    }
+
+    public void appendArtifacts(List<Artifact> artifacts, String moduleId) {
+        Module defaultModule = new ModuleBuilder()
+                .id(moduleId)
+                .artifacts(artifacts)
+                .build();
+        Module currentModule = this.getModules().stream()
+                // Check if the default module already exists.
+                .filter(module -> StringUtils.equals(module.getId(), moduleId))
+                .findAny()
+                .orElse(null);
+        if (currentModule != null) {
+            currentModule.append(defaultModule);
+        } else {
+            this.getModules().add(defaultModule);
+        }
```

Comment:
The above code, which finds the current module and appends it is duplicated twice in this class. Let's move it into a private method named ```addDefaultModule()``` and reuse it.
This comment referes to the following code:
```
        Module currentModule = this.getModules().stream()
                // Check if the default module already exists.
                .filter(module -> StringUtils.equals(module.getId(), moduleId))
                .findAny()
                .orElse(null);
        if (currentModule != null) {
            currentModule.append(defaultModule);
        } else {
            this.getModules().add(defaultModule);
        }
```